pkg/operator/genericoperatorclient/dynamic_operator_client.go

```diff
@@ -47,6 +47,36 @@ func (c dynamicOperatorClient) Informer() cache.SharedIndexInformer {
 	return c.informer.Informer()
 }
 
+func (c dynamicOperatorClient) GetObjectMeta() (*metav1.ObjectMeta, error) {
+	uncastInstance, err := c.informer.Lister().Get(globalConfigName)
+	if err != nil {
+		return nil, err
+	}
+	instance := uncastInstance.(*unstructured.Unstructured)
+	return getObjectMetaFromUnstructured(instance.UnstructuredContent())
+}
+
+func (c dynamicOperatorClient) UpdateObjectMeta(resourceVersion string, meta *metav1.ObjectMeta) (*metav1.ObjectMeta, error) {
+	uncastOriginal, err := c.informer.Lister().Get(globalConfigName)
+	if err != nil {
+		return nil, err
+	}
+	original := uncastOriginal.(*unstructured.Unstructured)
+
+	copy := original.DeepCopy()
+	copy.SetResourceVersion(resourceVersion)
```

Comment:
this way we risk to lose data from the spec. For the other two updaters this is slightly different as the status and the spec are the input and the operator claims ownership of these values.

Reply:
That's fair, I hadn't put too much thought on this one.

It turns out I don't need this method, so I dropped it. I just need to read the metadata.